packages/frontend/src/components/Modal/index.js

```diff
@@ -1,10 +1,28 @@
 /* eslint-disable jsx-a11y/media-has-caption */
-import React from 'react';
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+function Modal({ setShowModal, setShowVideoPlayer }) {
+  const modalContentRef = useRef();
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (modalContentRef.current && !modalContentRef.current.contains(event.target)) {
+        setShowModal(false);
+        setShowVideoPlayer(true);
+      }
+    };
```

Comment:
One recommendation on code structure — I would create a callback function in `<Home>` that does this, and pass that callback into to `<Modal>`  to be called here. That way the knowledge of how to hide and show the modal would be contained in one place. So this would become

```suggestion
    const handleClickOutside = (event) => {
      if (modalContentRef.current && !modalContentRef.current.contains(event.target)) {
        handleClickOutside();
      }
    };
```

and you'd pass it in like

```js
<Modal handleClickOutside={() => {
  setShowModal(false);
  setShowVideoPlayer(true);
}} />

Reply:
(This would also let you restart the reset timer when the modal gets closed)